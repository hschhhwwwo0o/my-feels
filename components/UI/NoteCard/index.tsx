import React, { FunctionComponent } from "react";
import Link from "next/link";

const NoteCard: FunctionComponent = () => {
  return (
    <Link href="/note/1">
      <a>
        <div className="cursor-pointer">
          <div className="border-2 border-[#EAEAEA] dark:border-[#2D2D2D] w-full max-w-[164px] lg:max-w-[272px] h-[164px] lg:h-[272px] rounded-[21px] shadow-md px-4 lg:px-6 py-4 lg:py-5">
            <div>
              <h2 className="font-medium text-sm lg:text-lg text-ellipsis overflow-hidden whitespace-nowrap text-[#242424] dark:text-[#D8D8D8]">
                My Diary Note
              </h2>
            </div>
            <div className="mt-2 lg:mt-3">
              <p className="text-[#878787] dark:text-[#A5A5A5] text-[10px] lg:text-sm max-h-[73px] lg:max-h-[140px] overflow-hidden text-justify font-medium">
                â€œArchive is the publishing house behind the most inspiring books all around Europe. We designed the
                Archive website with some exciting features to give users an even better access to their goodies and we
                also made an improved mobile version.
              </p>
            </div>
            <div>
              <div className="text-[10px] lg:text-xs text-[#C5C5C5] dark:text-[#A5A5A5] font-semibold mt-3 lg:mt-6">
                12/02/2022
              </div>
            </div>
          </div>
        </div>
      </a>
    </Link>
  );
};

export default NoteCard;
