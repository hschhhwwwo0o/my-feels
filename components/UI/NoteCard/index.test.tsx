import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import NoteCard from "./index";

test("Test NoteCard component", () => {
  render(
    <span data-testid="card">
      <NoteCard />
    </span>
  );
  const el: HTMLElement = screen.getByTestId("card");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
