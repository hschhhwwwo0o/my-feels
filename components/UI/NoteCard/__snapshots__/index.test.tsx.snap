// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NoteCard component 1`] = `
<span
  data-testid="card"
>
  <div
    style="opacity: 0;"
  >
    <div
      class="cursor-pointer"
    >
      <div
        class="border-2 border-[#EAEAEA] hover:border-[#d6d6d6] dark:border-[#2D2D2D] dark:hover:border-[#5f5f5f] w-full max-w-[164px] lg:max-w-[272px] h-[164px] lg:h-[272px] rounded-[21px] shadow-md hover:shadow-xl px-4 lg:px-6 py-4 lg:py-5 transition-all duration-1000"
      >
        <div>
          <h2
            class="font-medium text-sm lg:text-lg text-ellipsis overflow-hidden whitespace-nowrap text-[#242424] dark:text-[#D8D8D8] transition-all duration-1000"
          />
        </div>
        <div
          class="mt-2 lg:mt-3"
        >
          <p
            class="text-[#878787] dark:text-[#A5A5A5] text-[10px] lg:text-sm min-h-[73px] max-h-[73px] lg:min-h-[140px] lg:max-h-[140px] overflow-hidden text-justify font-medium transition-all duration-1000"
          >
            The note contains nothing
          </p>
        </div>
        <div>
          <div
            class="text-[10px] lg:text-xs text-[#C5C5C5] dark:text-[#A5A5A5] font-semibold mt-3 lg:mt-6 transition-all duration-1000"
          >
            Invalid date
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
